README.md

```diff
@@ -1,43 +1,50 @@
 # CPP-GL
 
+General purpose header-only template graph library for C++20 and newer standards.
+
 [![g++](https://github.com/SpectraL519/cpp-gl/actions/workflows/gpp.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/g++)
 [![clang++](https://github.com/SpectraL519/cpp-gl/actions/workflows/clang.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/clang++)
 [![format](https://github.com/SpectraL519/cpp-gl/actions/workflows/format.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/format)
 
-<!-- TODO: modify master branch protection ruleset (require all checks to pass) -->
-
 <br />
 
 ## Overview
-`CPP-GL` is a general purpose header-only template graph library for C++20 and newer standards.
 
-<br />
+- The goal of the project was to create a higlhy customizable, intuitive and easy to work with graph library for the modern C++ standards.
+- The `CPP-GL` library's implementation relies solely on the C++ standard library (it does not require installing any additional tools) and is designed to be compatible with the C++ standard tools, e.g. range-based loops, std algorithms, the [ranges library](https://en.cppreference.com/w/cpp/ranges), stream operations, etc.
+- The library relies heavily on [concepts](https://en.cppreference.com/w/cpp/language/constraints) to acheive abstraction instead of interfaces and abstract classes to minimize the overhead associated with virtual tables and dynamic dispatch.
 
-## Table of contents
-* [Tutorial](#tutorial)
-    * [Including CPP-GL into a project](#including-cpp-gl-into-a-project)
-        * [CMake integration](#cmake-integration)
-        * [Downloading the library](#downloading-the-library)
-    * ...
-* Examples
-* [Dev notes](#dev-notes)
-    * [Building and testing](#building-and-testing)
-    * [Formatting](#formatting)
-* Compiler support
-* [Licence]()
+> [!NOTE]
```

Comment:
Modify this note to say that the v1.0.* and the prereleases were developed as a thesis